pika/channel.py

```diff
@@ -753,127 +756,132 @@ def open(self):
 
-        :param callable callback: callback(pika.frame.Method) for method
-          Queue.DeclareOk; MUST be None when nowait=True.
         :param queue: The queue name
         :type queue: str or unicode
         :param bool passive: Only check to see if the queue exists
         :param bool durable: Survive reboots of the broker
         :param bool exclusive: Only allow access by the current connection
         :param bool auto_delete: Delete after consumer cancels or disconnects
-        :param bool nowait: Do not wait for a Queue.DeclareOk
         :param dict arguments: Custom key/value arguments for the queue
+        :param callable callback: callback(pika.frame.Method) for method
+          Queue.DeclareOk
+        :raises ValueError:
 
         """
+        self._validate_channel()
+        nowait = self._validate_rpc_completion_callback(callback)
+
         if queue:
-            condition = (spec.Queue.DeclareOk,
-                         {'queue': queue})
+            condition = (spec.Queue.DeclareOk, {'queue': queue})
         else:
-            condition = spec.Queue.DeclareOk  # pylint: disable=R0204
-        replies = [condition] if nowait is False else []
-        self._validate_channel_and_callback(callback)
+            condition = spec.Queue.DeclareOk
+        replies = [condition] if not nowait else []
+
         return self._rpc(spec.Queue.Declare(0, queue, passive, durable,
                                             exclusive, auto_delete, nowait,
                                             arguments or dict()),
                          callback, replies)
 
     def queue_delete(self,
-                     callback=None,
                      queue='',
                      if_unused=False,
                      if_empty=False,
-                     nowait=False):
+                     callback=None):
         """Delete a queue from the broker.
 
-        :param callable callback: The callback to call on Queue.DeleteOk;
-            MUST be None when nowait=True.
         :param queue: The queue to delete
         :type queue: str or unicode
         :param bool if_unused: only delete if it's unused
         :param bool if_empty: only delete if the queue is empty
-        :param bool nowait: Do not wait for a Queue.DeleteOk
+        :param callable callback: The callback to call on Queue.DeleteOk
+        :raises ValueError:
 
         """
-        replies = [spec.Queue.DeleteOk] if nowait is False else []
-        self._validate_channel_and_callback(callback)
+        self._validate_channel()
+        nowait = self._validate_rpc_completion_callback(callback)
+        replies = [spec.Queue.DeleteOk] if not nowait else []
         return self._rpc(spec.Queue.Delete(0, queue, if_unused, if_empty,
                                            nowait),
                          callback, replies)
 
-    def queue_purge(self, callback=None, queue='', nowait=False):
+    def queue_purge(self, queue='', callback=None):
```

Comment:
`queue` shouldn't have a default value since you always have to name the queue to be purged